### DISCUSSION TITLE
[Enhancement] The most wanted things!!!

### DISCUSSION THREAD

**Discussion by mhrastegari** [dotnet/maui#76] [General]
## Summary
If MAUI is gonna be something cooler, it should have the most wanted things that developers wants from Xamarin.Forms and never happened...

## Some requests from Xamarin.Forms

- [Popup Control](https://github.com/xamarin/Xamarin.Forms/issues/1778)
- [Context Menu](https://github.com/xamarin/Xamarin.Forms/issues/5883) and [Context Flyout](https://github.com/xamarin/Xamarin.Forms/issues/10372)
- [Shadows](https://github.com/xamarin/Xamarin.Forms/issues/7744)
- [Transitions Api](https://github.com/xamarin/Xamarin.Forms/issues/6033)
- [SegmentsView Control](https://github.com/xamarin/Xamarin.Forms/issues/6904)
- [ComboBox And AutoSuggestBox](https://github.com/xamarin/Xamarin.Forms/issues/1775)
- [Content Button](https://github.com/xamarin/Xamarin.Forms/issues/6872)
- [Floating Action Button](https://github.com/xamarin/Xamarin.Forms/issues/1714)
- [Set StatusBar color for iOS and Android from any Page](https://github.com/xamarin/Xamarin.Forms/issues/7314)

No proposal founded:

- MicroCharts ([Like Javier Suárez's Twitter post](https://twitter.com/jsuarezruiz/status/1274745705362608129?s=20))
- **Bottom Sheet**
- CrossPlatform SplashScreen
- CrossPlatform App Logo
- Native Page Transitions
- Parallax Effect
- Transition Collection like in [UWP](https://www.google.com/url?sa=t&source=web&rct=j&url=https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.animation.transitioncollection%3Fview%3Dwinrt-18362&ved=2ahUKEwiA5sjkwsnpAhWPsJ4KHXjlBn8QFjAAegQIAhAB&usg=AOvVaw0T2hdzppaCL7AOU0Ij1_8S)
- Better CornerRadius: (Setting top-left, top-right, bottom-left, bottom-right)
- Customizable Entry

 ## What i Really wish

[lowering UWP minimum requirement to 10240](https://github.com/xamarin/Xamarin.Forms/issues/8570) (or at least 15063 and 14393)

**this list will be updated...**

**Comment by Happypig375** [comment]
Add all of https://github.com/xamarin/Xamarin.Forms/labels/F100 to this list, thanks

**Comment by qtbui159** [comment]
Web support.  Convert xaml to html+css+js or webassembly

**Comment by dpaulino** [comment]
@qtbui159 this is off topic, but Uno already brings xaml to web using webassembly https://github.com/unoplatform/uno

**Comment by Happypig375** [comment]
@dpaulino The real Xamarin.Forms on the web is https://github.com/praeclarum/Ooui and I hope that it can be officially absorbed into Xamarin.Forms/Maui.

**Comment by mgierlasinski** [comment]
@mhrastegary77 Floating Action Button and Extended Floating Action Button:
https://github.com/xamarin/Xamarin.Forms/issues/1714

**Comment by qtbui159** [comment]
@dpaulino I mean the problem of MAUI with the current plan is we still have web dev and mobile/desktop dev separate from each other.

**Comment by richardvaldiviesomacias** [comment]
It would be great to add a bottom sheet component (Floating View..maybe) It is always a pain to do something like that from scratch.  There are nuget packages, but not all of them are smooth enough in both Android and iOS.

**Comment by AmrAlSayed0** [comment]
While all that is mentioned here is great, I would really welcome some kind for feature parity with the CSS Box Model. Let me explain.

Any normal HTML element can have its padding, margin, border, corners, shadows and background customized and I really mean ANY HTML element. Also, you can choose the kind of box sizing (eg. content box, border-box, padding-box?) I would really like to see all of these implemented for ALL Xamarin.Forms/MAUI views including layouts like Grid and Stack.

There are some crazy things in CSS that would be really difficult to implement for mobile like ```border-image``` for example so not all properties have to be implemented. But it's not unreasonable for example to want a Background property that can support colors, images and gradients or a Border property for applying rounded colored borders, bonus points if each border can be customized separately.

I would also very mush like to have the ability to add relative values. There a lot of times where I wished I could add Padding to the sides of a view that is relative to the width of the view. Doable now through converters but still a hassle to get it working.

If all of these CORE features are implemented it would give developers a lot of flexibility to customize the UI and there won't be a need for external views like Pancake view (which doesn't really apply the borders or shadows to the view itself, it just wraps and then clips the inner view) but that doesn't mean I am not grateful for the author of that library 😅.

**Comment by ivandharo** [comment]
we really need a bottom action sheet like material design :(

**Comment by mhrastegari** [comment]
@ivanbasov we really don't have much menus in Xamarin.Forms we just have picker and action sheet! no context menus, no context flyouts, no popups, nor dropDown menus and no bottom sheets.
even the design of lisview contextMenu is outdated on iOS and is kinda a SwipeView Duplicate plus still no icons support for iOS and UWP!

I hope MAUI will surprise us all :(  ❤

**Comment by mrwcjoughin** [comment]
there are icons if you use font awesome ? it works very well and even better scales automatically as its a font!

**Comment by mhrastegari** [comment]
@mrwcjoughin sorry i didn't write that correctly! i was saying listview ContextMenu doesn't support icons for UWP and iOS :)

**Comment by FruitTreeAlchemist** [comment]
Better support for text trimming and wrapping.

Some Xaml goodies, First-class support for VisualStateManager, data/event triggers, and behaviors. If possible, control templates would be amazing.

Performance on Android needs some help. Android with views/pages vs IOS is not comparable.  However, performance is being looked at in MAUI.

CollectionView renamed to ItemsControl? Might be confused with System.Windows.Data.CollectionView ?

**Comment by Xyncgas** [comment]
And make sure 1 pixel is exactly 1/72nd of an inch
Because regardless of screen size people wants to look at the same size of text unless they are special
